build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/tscTask.ts

```diff
@@ -485,6 +485,7 @@ export abstract class TscDependentTask extends LeafWithDoneFileTask {
 				const tsBuildInfo = await dep.readTsBuildInfo();
 				if (tsBuildInfo === undefined) {
 					// If any of the tsc task don't have build info, we can't track
+					this.traceError(`tsbuildinfo file not found for ${dep.name}: ${dep.command}`);
```

Comment:
This is TscDependentTask; so, it is indeed incorrect to have something depend on a tsc task that doesn't emit. Seems helpful to help out by checking the noEmit situation if possible. If not, then would be helpful to emit that as something to check for. If neither, we should at the very least add comments about the known situation here.